### PR TITLE
[ examples] fix loading jsonl with load dataset in run translation example

### LINKED ISSUE #26868: Dataset 'jsonl' doesn't exist on the Hub in examples/pytorch/translation/run_translation.py
### System Info

- `transformers` version: 4.35.0.dev0
- Platform: Linux-6.2.0-34-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.3
- Accelerate version: 0.24.0.dev0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.1.0+cu121 (True)
- Tensorflow version (GPU?): 2.10.0 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@patil-suraj

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

To reproduce the issue, follow the steps outlined below:

1. Environment Setup:

- Python >= 3.10
- pip install datasets

2. Dataset Files:

Have train.jsonl and validation.jsonl files present at the paths specified.

3. Code to Reproduce the Issue:

```python
Copy code
import datasets

# Setting data file paths
data_files = {}
data_files["train"] = "/home/mathias/translation_scripts/data/opus/huggingface/train_opus_en-da.jsonl"
data_files["validation"] = "/home/mathias/translation_scripts/data/opus/huggingface/validation_opus_en-da.jsonl"
print(data_files)

# Loading the dataset
from datasets import load_dataset
load_dataset("jsonl", data_files=data_files)
```
4. Error Message:

When running the code above, the following error message is generated:

```bash
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mathias/.local/lib/python3.10/site-packages/datasets/load.py", line 2129, in load_dataset
    builder_instance = load_dataset_builder(
  File "/home/mathias/.local/lib/python3.10/site-packages/datasets/load.py", line 1815, in load_dataset_builder
    dataset_module = dataset_module_factory(
  File "/home/mathias/.local/lib/python3.10/site-packages/datasets/load.py", line 1508, in dataset_module_factory
    raise FileNotFoundError(
FileNotFoundError: Couldn't find a dataset script at /home/mathias/translation_scripts/jsonl/jsonl.py or any data file in the same directory. Couldn't find 'jsonl' on the Hugging Face Hub either: FileNotFoundError: Dataset 'jsonl' doesn't exist on the Hub. If the repo is private or gated, make sure to log in with `huggingface-cli login`.
```

### Expected behavior

With the following python code:

```python
data_files = {}
data_files["train"] = "path/to/train.jsonl"
data_files["validation"] = "path/to/validation.jsonl"

load_dataset("jsonl", data_files=data_files)
```

The following output should be expected:

```bash
Downloading data files: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 29746.84it/s]
Extracting data files: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 1740.74it/s]
Generating train split: 1000000 examples [00:00, 3119077.05 examples/s]
Generating validation split: 2000 examples [00:00, 1100578.33 examples/s]
DatasetDict({
    train: Dataset({
        features: ['translation'],
        num_rows: 1000000
    })
    validation: Dataset({
        features: ['translation'],
        num_rows: 2000
    })
})
```

### PR DESCRIPTION
**Pull request by mathiasesn**
# What does this PR do?

Fixes #26868 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
LGTM, this might need to be propagated to other examples? Also can you update the pr description to say what this fixes and the name of the PR with [ `examples`] fix blablalbla

File: examples/pytorch/translation/run_translation.py [hunk 1]
```diff
@@ -364,15 +364,17 @@ def main():
...
         if data_args.validation_file is not None:
             data_files["validation"] = data_args.validation_file
-            extension = data_args.validation_file.split(".")[-1]
+            builder_name = data_args.validation_file.split(".")[-1]
         if data_args.test_file is not None:
             data_files["test"] = data_args.test_file
-            extension = data_args.test_file.split(".")[-1]
+            builder_name = data_args.test_file.split(".")[-1]
+        if builder_name == "jsonl":
+            builder_name = "json"
```
Comment (lhoestq) [inline examples/pytorch/translation/run_translation.py:375]: (nit) maybe clearer this way
```suggestion
        if extension == "jsonl":
            builder_name = "json"  # the "json" builder reads both .json and .jsonl files
        else:
            builder_name = extension  # e.g. "parquet"
```

File: examples/pytorch/translation/run_translation.py [hunk 2]
```diff
@@ -364,15 +364,17 @@ def main():
         data_files = {}
         if data_args.train_file is not None:
             data_files["train"] = data_args.train_file
-            extension = data_args.train_file.split(".")[-1]
+            builder_name = data_args.train_file.split(".")[-1]
         if data_args.validation_file is not None:
             data_files["validation"] = data_args.validation_file
-            extension = data_args.validation_file.split(".")[-1]
+            builder_name = data_args.validation_file.split(".")[-1]
```
Comment (lhoestq) [inline examples/pytorch/translation/run_translation.py:370]:
```suggestion
            extension = data_args.validation_file.split(".")[-1]
```

File: examples/pytorch/translation/run_translation.py [hunk 3]
```diff
@@ -364,15 +364,17 @@ def main():
         data_files = {}
         if data_args.train_file is not None:
             data_files["train"] = data_args.train_file
-            extension = data_args.train_file.split(".")[-1]
+            builder_name = data_args.train_file.split(".")[-1]
```
Comment (lhoestq) [inline examples/pytorch/translation/run_translation.py:367]:
```suggestion
            extension = data_args.train_file.split(".")[-1]
```

File: examples/pytorch/translation/run_translation.py [hunk 4]
```diff
@@ -364,15 +364,17 @@ def main():
...
         if data_args.validation_file is not None:
             data_files["validation"] = data_args.validation_file
-            extension = data_args.validation_file.split(".")[-1]
+            builder_name = data_args.validation_file.split(".")[-1]
         if data_args.test_file is not None:
             data_files["test"] = data_args.test_file
-            extension = data_args.test_file.split(".")[-1]
+            builder_name = data_args.test_file.split(".")[-1]
```
Comment (lhoestq) [inline examples/pytorch/translation/run_translation.py:373]:
```suggestion
            extension = data_args.test_file.split(".")[-1]
```

**Review by lhoestq — APPROVED** [review]
Thanks !

I added a minor suggestion

**Comment by mathiasesn** [comment]
> LGTM, this might need to be propagated to other examples? Also can you update the pr description to say what this fixes and the name of the PR with [ `examples`] fix blablalbla

Update name of the PR to: [ examples] fix loading jsonl with load dataset in run translation example.

**Review by ArthurZucker — APPROVED** [review]
Thanks for the nit

**Comment by ArthurZucker** [comment]
Thanks for the fix!

**Pull request merged**